index.bs

```diff
@@ -575,79 +575,225 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
+			typedef (DOMString or Blob) ClipboardItemDataType;
+			typedef Promise&lt;ClipboardItemDataType&gt; ClipboardItemData;
 
-		[SecureContext, Exposed=Window] interface Clipboard : EventTarget {
-		  Promise&lt;ClipboardItems> read();
-		  Promise&lt;DOMString> readText();
-		  Promise&lt;undefined> write(ClipboardItems data);
-		  Promise&lt;undefined> writeText(DOMString data);
-		};
+			[SecureContext, Exposed=Window]
+			interface ClipboardItem {
+				constructor(record&lt;DOMString, ClipboardItemData> items,
+				            optional ClipboardItemOptions options = {});
+
+				readonly attribute PresentationStyle presentationStyle;
+				readonly attribute FrozenArray&lt;DOMString> types;
+
+				Promise&lt;Blob&gt; getType(DOMString type);
+			};
 
-		typedef (DOMString or Blob) ClipboardItemDataType;
-		typedef Promise&lt;ClipboardItemDataType> ClipboardItemData;
+			enum PresentationStyle { "unspecified", "inline", "attachment" };
 
-		callback ClipboardItemDelayedCallback = ClipboardItemData ();
+			dictionary ClipboardItemOptions {
+				PresentationStyle presentationStyle = "unspecified";
+			};
+		</pre>
 
-		[Exposed=Window] interface ClipboardItem {
-		  constructor(record&lt;DOMString, ClipboardItemData> items,
-		    optional ClipboardItemOptions options = {});
-		  static ClipboardItem createDelayed(
-		      record&lt;DOMString, ClipboardItemDelayedCallback> items,
-		      optional ClipboardItemOptions options = {});
+		<dl class=note>
+			<dt><code><var>clipboardItem</var> = new ClipboardItem([<var>items</var>, <var>options</var>])</code>
+			<dd>
+			Creates a new {{ClipboardItem}} object. <var>items</var> are used to fill its MIME types and [=Promise=]s to [=Blob=]s or {{DOMString}}s corresponding to the MIME types, <var>options</var> can be used to fill its {{ClipboardItemOptions}},
+			as per the example below.
 
-		  readonly attribute PresentationStyle presentationStyle;
-		  readonly attribute long long lastModified;
-		  readonly attribute boolean delayed;
+				<pre class="example javascript" highlight=js>
+					const format1 = 'text/plain';
+					const promise_text_blob = Promise.resolve(new Blob(['hello'], {type: format1}));
+					const clipboardItemInput = new ClipboardItem(
+						{[format1]: promise_text_blob},
+						{presentationStyle: "unspecified"});
+				</pre>
 
-		  readonly attribute FrozenArray&lt;DOMString> types;
+			<dt><code><var>clipboardItem</var>.getType(<var>type</var>)</code>
+			<dd><p>Returns a [=Promise=] to the [=Blob=] corresponding to <var>type</var>.</p>
 
-		  Promise&lt;Blob> getType(DOMString type);
-		};
+			<dt><code><var>clipboardItem</var>.<var>types</var></code>
+			<dd><p>Returns the list of <var>types</var> contained in the <var>clipboardItem</var> object.
+
+		</dl>
+
+		<h4 id="clipboard-item">Clipboard Item</h4>
+
+		A [=clipboard item=] is conceptually data that the user has expressed a desire to make shareable by invoking a "cut" or "copy" command.
```

Comment:
This is only half the story. `clipboard item`s can also stem from "paste" operations or from websites creating them. It seems that information is relevant for web-developers, so shouldn't it be encapsulated in a `domintro` class for the IDL?

Reply:
This is the conceptual `Clipboard Item` object which only makes sense to create if user has initiated a cut or copy operation. Paste wouldn't do anything if there is nothing on the clipboard. Websites would use the `{{ClipboardItem}}` object, not the conceptual `Clipboard Item`.

Reply:
> Paste wouldn't do anything if there is nothing on the clipboard.

If there's something on the system clipboard, pasting it to the browser would create a `ClipboardItem`, which, according to section 7.2.1 has an associated clipboard item: "A ClipboardItem object has an associated clipboard item, which is a clipboard item."

Reply:
Perhaps it makes sense to define the conceptual "clipboard item" as a struct, analogous to "fetch params" defined at https://fetch.spec.whatwg.org/#fetch-params. I.e., `A clipboard item is a a struct ...  It has the following items ...`.

Reply:
Added `<dfn>` for `clipboard item` as suggested by @domenic .